File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    total = brown + yellow
    answer = []
    
    for x in range(1,int(yellow**0.5)+1):
        if yellow % x == 0:
            y = yellow //x 
            if 2*(x+y+2) == brown:
                answer.append(y+2)

                answer.append(x+2)
        
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let total = brown + yellow
    var answer = [];
    for (let x =1; x<=(yellow)**0.5; x++){
        if (yellow%x === 0){
            let y = yellow/x
            if ((x+y+2)*2 === brown){
                answer.push(y+2)
                answer.push(x+2)
            }
        }
    }
    return answer;
}